centreon/src/Core/Dashboard/Infrastructure/API/PartialUpdateDashboard/PartialUpdateDashboardController.php

```diff
@@ -53,101 +65,89 @@ final class PartialUpdateDashboardController extends AbstractController
     public function __invoke(
         int $dashboardId,
         Request $request,
+        #[MapRequestPayload] PartialUpdateDashboardRequest $mappedRequest,
         PartialUpdateDashboard $useCase,
         PartialUpdateDashboardPresenter $presenter,
     ): Response {
         $this->denyAccessUnlessGrantedForApiConfiguration();
 
         try {
-            $partialUpdateDashboardRequest = $this->getPartialUpdateDashboardRequest($request);
+            $partialUpdateDashboardRequest = $mappedRequest->toDto();
+
+            $this->assertThumbnailDataSent($request, $partialUpdateDashboardRequest);
+
+            if (
+                $request->files->get('thumbnail_data') !== null
+                && ! $partialUpdateDashboardRequest->thumbnail instanceof NoValue
+            ) {
+                /** @var UploadedFile $thumbnail */
+                $thumbnail = $request->files->get('thumbnail_data');
+                $partialUpdateDashboardRequest->thumbnail->content = $this->validateAndRetrieveThumbnailContent(
+                    $thumbnail,
+                );
+            }
```

Comment:
IMHO `assertThumbnailDataSent` method could be avoided:
- It will lead to handle the missing file **differently** than any other missing input
- You can use **validation** against a null value if not present in the request, leading to a classic 422. You will need to add the NotBlank constraint below and that's it.

```suggestion
            if (! $partialUpdateDashboardRequest->thumbnail instanceof NoValue) {
                $partialUpdateDashboardRequest->thumbnail->content = $this->validateAndRetrieveThumbnailContent(
                    $request->files->get('thumbnail_data', null),
                );
```

Reply:
You are right, changes applied